File: Tikraq/src/main/java/com/example/tikraq/entities/Suscripcion.java
package com.example.tikraq.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@Entity
@Table(name = "suscripciones")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Suscripcion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String tipo;
    private LocalDate fechaInicio;
    private LocalDate fechaFin;
    private String estado;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "usuarios_usuarioid")
    private Usuario usuariosUsuarioid;

}

File: TikraqApp/src/app/model/Suscripcion.ts
export class Suscripcion {
  id: number;
  tipo: string;
  fechaInicio: Date;
  fechaFin: Date;
  estado: string;
  usuariosUsuarioid: { id:number };

  constructor(id: number, tipo: string, fechaInicio: Date, fechaFin: Date, estado: string, usuariosUsuarioid: {id:number }) {
    this.id = id;
    this.tipo = tipo;
    this.fechaInicio = fechaInicio;
    this.fechaFin = fechaFin;
    this.estado = estado;
    this.usuariosUsuarioid = usuariosUsuarioid;
  }
}
